### DISCUSSION TITLE
How to correctly convert code blocks with newlines?

### DISCUSSION THREAD

**Discussion by leakec** [facelessuser/pymdown-extensions#1716] [Q&A] [answered]
Hello, 

I'm having trouble correctly converting the following code block, i.e., with a newline at the beginning and the end:
````
```

code-block

```
````
I'm trying to do this conversion using
```
python -m markdown test.md -x pymdownx.extra
```
where `test.md` contains the code block above.

If I run  it using a code block without spaces, i.e.,

````
```
code-block
```
````

everything runs correctly. Alternatively, if I downgrade `pymdown-extensions` to version 9.2, the code block with newlines runs correctly. Is this a bug, or is there something I need to do in version 9.4 to get code blocks with newlines to convert correctly? 

Thanks,
Carl

**Comment by facelessuser** [comment, marked as answer]
I'm not sure why you are having issues. So I checked out the latest tag (9.4). Dropped your markdown into a `test.md` file, and this is what I got:

````console
➜  pymdown-extensions git:(72390ce2) ✗ python3 -m markdown -x pymdownx.extra example.md
<div class="highlight"><pre><span></span><code>code-block
</code></pre></div>%
````

Can you elaborate on what you mean my it isn't working correctly?

**Reply by leakec** [reply]
When I run the same thing I get
```
<p>```</p>
<p>code-block</p>
<p>```</p>%
```
which doesn't contain any `<code>` HTML tags.

If I remove the newlines in the code block I get:
```
<p><code>code-block</code></p>%
```
as expected.

For reference I am running Python 3.9.12 on Fedora 34. Other package versions I think might be relevant are `Markdown` at version `3.3.7`. Please let me know if there is any other information that I can provide that will help.

**Reply by facelessuser** [reply]
I've created an isolated environment, I cannot reproduce your issue. I imagine you have something wrong locally in your environment. Can you try and create a virtual environment that isolates the test as I've done below?

````
➜  pymdownx-test python3 -m venv venv
➜  pymdownx-test source venv/bin/activate
(venv) ➜  pymdownx-test pip install markdown pymdown-extensions
Collecting markdown
  Using cached Markdown-3.3.7-py3-none-any.whl (97 kB)
Collecting pymdown-extensions
  Using cached pymdown_extensions-9.4-py3-none-any.whl (216 kB)
Installing collected packages: markdown, pymdown-extensions
Successfully installed markdown-3.3.7 pymdown-extensions-9.4
WARNING: You are using pip version 22.0.4; however, version 22.1.1 is available.
You should consider upgrading via the '/Users/facelessuser/Code/pymdownx-test/venv/bin/python3 -m pip install --upgrade pip' command.
(venv) ➜  pymdownx-test echo "\`\`\`\n\ncode-block\n\n\`\`\`" > test.md
(venv) ➜  pymdownx-test cat test.md
```

code-block

```
(venv) ➜  pymdownx-test python3 -m markdown -x pymdownx.extra test.md
<pre class="highlight"><code>
code-block
</code></pre>%
````

**Reply by leakec** [reply]
You're right, creating a new isolated enviornment fixed the issue (see output below):
```
[leake/Downloads] $ python3 -m venv venv
[leake/Downloads] $ source venv/bin/activate
(venv) [leake/Downloads] $ pip install markdown pymdown-extensions
Collecting markdown
  Using cached Markdown-3.3.7-py3-none-any.whl (97 kB)
Collecting pymdown-extensions
  Using cached pymdown_extensions-9.4-py3-none-any.whl (216 kB)
Collecting importlib-metadata>=4.4
  Downloading importlib_metadata-4.11.4-py3-none-any.whl (18 kB)
Collecting zipp>=0.5
  Downloading zipp-3.8.0-py3-none-any.whl (5.4 kB)
Installing collected packages: zipp, importlib-metadata, markdown, pymdown-extensions
Successfully installed importlib-metadata-4.11.4 markdown-3.3.7 pymdown-extensions-9.4 zipp-3.8.0
WARNING: You are using pip version 21.0.1; however, version 22.1.1 is available.
You should consider upgrading via the '/home/leake/Downloads/venv/bin/python3 -m pip install --upgrade pip' command.
(venv) [leake/Downloads] $ echo "\`\`\`\n\ncode-block\n\n\`\`\`" > test.md
(venv) [leake/Downloads] $ cat test.md
```

code-block

```
(venv) [leake/Downloads] $ python3 -m markdown -x pymdownx.extra test.md
<pre class="highlight"><code>
code-block
</code></pre>%
```

I've changed my regular environment to have all of the same versions as the virtual environment, but am still seeing the issue. There must be a package in my regular environment conflicting. I will investigate and post back if I find anything. Thank you for all of your help :)

**Reply by facelessuser** [reply]
No problem. If I'm ever confused about why things aren't working as documented, isolating the environment to the bare minimum is often quite helpful. Hopefully, you'll be able to figure out what's going on 🙂.

**Reply by facelessuser** [reply]
Hmm. Maybe update Pygments? It's the only other thing I can think may be giving you trouble.

**Reply by leakec** [reply]
I'm not sure which package was the offending one, but updating everything via:
```
pip list --outdated --format=freeze | grep -v '^\-e' | cut -d = -f 1  | xargs -n1 pip install -U
```
did the trick. Now everything woks with version 9.4 and 9.2 :) 

Thanks for all the help!

**Reply by leakec** [reply]
You're right `pygments` was it. I tried downgrading to the one I originally had (2.7.4), and the issue popped up again. Upgrading it to 2.12.0 fixes it!

**Reply by facelessuser** [reply]
Okay, that makes sense. Pygments made a change to their formatting very recently. Pymdown Extensions had to accommodate it properly in the most recent version. Sorry I didn't put 2 + 2 together on that one earlier 🙃.

**Reply by leakec** [reply]
No worries; I'm glad you found it, since there a few more computers this will be used on that will need to update pygments as well.
